internals/osutil/mkdirallchown_test.go

```diff
@@ -36,7 +36,7 @@ func (mkdacSuite) TestSlashySlashy(c *check.C) {
 		d := c.MkDir()
 		// just in case
 		c.Assert(strings.HasSuffix(d, "/"), check.Equals, false)
-		err := osutil.MkdirAllChown(d+dir, 0755, osutil.NoChown, osutil.NoChown)
+		err := osutil.MkdirAllChown(d+dir, 0755, osutil.AtomicWriteChmod, osutil.NoChown, osutil.NoChown)
```

Comment:
It seems to me we should pass 0 for the flags for all existing tests, so we're keeping the tests for that behaviour. And we should add new tests that test the new chmod flag / behaviour.